### DISCUSSION TITLE
How to solve nchat libwmchat library linking error?

### DISCUSSION THREAD

**Discussion by ferrodvd** [d99kris/nchat#136] [Q&A]
Recently I updated Go library (latest 1.21.1) on my arm64 linux system and now  I have a linking error with libwmchat.so library.
This is the output:

"[100%] Linking CXX shared library ../libwmchat.so
/usr/bin/ld: go/libcgowm.so(go.o): relocation R_AARCH64_ADR_PREL_PG_HI21 against symbol `crosscall2' which may bind externally can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: DWARF error: line info data is bigger (0x37080637) than the space remaining in the section (0xe0e72)
/usr/bin/ld: DWARF error: line info data is bigger (0x6370683) than the space remaining in the section (0xdf216)
/usr/bin/ld: DWARF error: line info data is bigger (0x37063806) than the space remaining in the section (0xda650)
/usr/bin/ld: DWARF error: line info data is bigger (0x56c00209) than the space remaining in the section (0xbe7b4)
/usr/bin/ld: DWARF error: unhandled .debug_line version 0
/usr/bin/ld: DWARF error: line info data is bigger (0x2f646c69) than the space remaining in the section (0x4f33a)
/usr/bin/ld: DWARF error: line info data is bigger (0x6d2f6372) than the space remaining in the section (0x369bc)
/usr/bin/ld: DWARF error: line info data is bigger (0xb400de03) than the space remaining in the section (0x2f77e)
go/libcgowm.so(go.o): in function `runtime/cgo.set_crosscall2.abi0':
go.go:(.text+0x292668): rilocazione pericolosa: unsupported relocation
collect2: error: ld returned 1 exit status
make[2]: *** [lib/wmchat/CMakeFiles/wmchat.dir/build.make:103: lib/libwmchat.so] Error 1
make[1]: *** [CMakeFiles/Makefile2:280: lib/wmchat/CMakeFiles/wmchat.dir/all] Error 2
make: *** [Makefile:136: all] Error 2"

I checked the -fPIC option presence in compile settings and it seems all ok.
Has anyone else detected this issue? Can you help me to solve it?

**Comment by d99kris** [comment]
Hi @ferrodvd - I suspect this is a golang bug - see https://github.com/golang/go/issues/62556 - perhaps you can try to switch back to an older go version until it's been fixed.

**Reply by ferrodvd** [reply]
ok, thank you very much for your suggestion.

**Reply by ferrodvd** [reply]
I downgraded golang to 1.20.9 and now the library gets compiled and linked correctly. Thanks again

**Reply by d99kris** [reply]
Ok good to hear the workaround worked 👍
